packages/react/hover-card/src/HoverCard.tsx

```diff
@@ -256,6 +263,50 @@ const HoverCardContentImpl = React.forwardRef<
   const popperScope = usePopperScope(__scopeHoverCard);
   const ref = React.useRef<HoverCardContentImplElement>(null);
   const composedRefs = useComposedRefs(forwardedRef, ref);
+  const [containTextSelection, setContainTextSelection] = React.useState(false);
+
+  React.useEffect(() => {
+    if (containTextSelection) {
+      const body = document.body;
+
+      // Safari requires prefix
+      originalBodyUserSelect = body.style.userSelect || body.style.webkitUserSelect;
+
+      body.style.userSelect = 'none';
+      body.style.webkitUserSelect = 'none';
+      return () => {
+        body.style.userSelect = originalBodyUserSelect;
+        body.style.webkitUserSelect = originalBodyUserSelect;
+      };
+    }
+  }, [containTextSelection]);
+
+  const { onDismiss } = context;
+
+  React.useEffect(() => {
+    if (ref.current) {
+      const handlePointerUp = (event: PointerEvent) => {
+        // Delay a frame to ensure we always access the latest selection
+        setTimeout(() => {
+          const hasEmptySelection = document.getSelection()?.toString() === '';
+          setContainTextSelection(false);
+
+          if (hasEmptySelection) {
+            const isPointerUpOutsideContent = !ref.current?.contains(event.target as HTMLElement);
```

Comment:
this won't respect the React tree? Do we care?

Reply:
I don't think we do, this is similar to the dismissible handling afaik which also uses `document`, is there even a way to handle that within the React tree?

This actually makes me wonder if we should be using `ownerDocument` here

Reply:
In DismissableLayer we do use the React tree right? for `onPointerDownOutside`/`onFocusOutside`.

Reply:
I thought internally it still needs to attach on `document`:
https://github.com/radix-ui/primitives/blob/5972f26d60dd1bb3b16888be4c9e3ab50e74aaf9/packages/react/dismissable-layer/src/DismissableLayer.tsx#L222

Perhaps I misunderstand how that functions?

Reply:
Thanks for clarifying this. I've implemented a similar approach but for `pointerUp` https://github.com/radix-ui/primitives/pull/1692/commits/d0bfcbd126f756e7100c5e82f7d082ab2e851b61